src/java_bytecode/ci_lazy_methods.cpp

```diff
@@ -477,17 +477,28 @@ void ci_lazy_methodst::gather_field_types(
   ci_lazy_methods_neededt &needed_lazy_methods)
 {
   const auto &underlying_type=to_struct_type(ns.follow(class_type));
-  for(const auto &field : underlying_type.components())
+  const typet &element_type=
```

Comment:
Prefer java_types.h `java_array_element_type` (with probably a check that is is an array).